plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,352 @@
+            }
+        }, indicesExistsResponseStepListener::onFailure);
+
+        isExistedIndex(ML_TASK_INDEX, indicesExistsResponseStepListener);
+
+        getReTryTimes(localNodeId, ActionListener.wrap(getReTryTimesStep::onResponse, getReTryTimesStep::onFailure));
+
+        Result result = new Result();
+        queryTaskStep.whenComplete(searchResponse -> {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return;
+            }
+            result.setHits(hits);
+
+            getReTryTimesStep.whenComplete(getReTryTimesResponse -> {
+                // if getReTryTimesResponse is null,it means we get reTryTimes at the first time,and the index
+                // .plugins-ml-model-reload doesn't exist,so we should let reTryTimes be zero(init value)
+                // we don't do anything
+                // if getReTryTimesResponse is not null,it means we have saved the value of reTryTimes into the index
+                // .plugins-ml-model-reload,so we get the value of the field MODEL_LOAD_RETRY_TIMES_FIELD
+                if (getReTryTimesResponse != null) {
+                    Map<String, Object> sourceAsMap = getReTryTimesResponse.getHits().getHits()[0].getSourceAsMap();
+                    result.setReTryTimes((Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD));
+                }
+
+                // According to the node id to get retry times, if more than the max retry times, don't need to retry
+                // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+                if (result.getReTryTimes() > autoReLoadMaxReTryTimes) {
+                    log.info("have exceeded max retry times, always failure");
+                    return;
+                }
+
+                int reTryTimes = 0;
+                try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, result.getHits()[0].getSourceRef())) {
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                    MLTask mlTask = MLTask.parse(parser);
+
+                    autoReLoadModelByNodeAndModelId(localNodeId, mlTask.getModelId());
+
+                    // if reload the model successfully,the number of unsuccessful reload should be reset to zero.
+                    result.setReTryTimes(reTryTimes);
+                } catch (RuntimeException e) {
+                    reTryTimes = result.getReTryTimes();
+                    reTryTimes++;
+                    result.setReTryTimes(reTryTimes);
+                    log.error("Can't auto reload model in node id {} ,has tried {} times\nThe reason is:{}", localNodeId, reTryTimes, e);
+                }
+
+                // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
```

Comment:
Any reason we have to persist retryTimes to index ? Is it ok to just cache retry times in memory?

Reply:
because when the ml-node happened reboot,the info of retrying time will be lost in cache if we put them in cache